views/install.ps1.erb

```diff
@@ -0,0 +1,191 @@
+function Install-ChefProject {
+  <#
+    .SYNOPSIS
+    Install Chef Client, Chef Development Kit, or AngryChef
+    .DESCRIPTION
+    Install Chef Client, Chef Development Kit, or AngryChef
+    .EXAMPLE
+    iex (new-object net.webclient).downloadstring('https:/www.chef.io/chef/install.ps1'); Install-ChefProject
+    
+    Installs the latest version of Chef Client.
+    .EXAMPLE
+    iex (irm 'https://www.chef.io/install.ps1'); Install-ChefProject -project ChefDK -p
+    
+    Installs the latest pre-release build of the Chef Development Kit
+  #>
+  [cmdletbinding()]
+  param (
+    # Version of the application to install
+    # This parameter is optional, if not supplied it will provide the latest version,
+    # and if an iteration number is not specified, it will grab the latest available iteration.
+    # Partial version numbers are also acceptable (using v=11 
+    # will grab the latest 11.x client which matches the other flags).
+    [string]
+    $version,
+    # Project to install
+    # chef - Chef Client
+    # chefdk - Chef Development Kit
+    # angrychef - AngryChef
+    # server and container are not valid windows targets
+    [validateset('chef', 'chefdk', 'angrychef')]
+    [string]
+    $project = 'chef',
+    # Full path for the downloaded installer.
+    [string]
+    $filename,
+    # Full path to the location to download the installer
+    [string]
+    $download_directory = $env:temp,
+    [switch]
+    $prerelease, 
+    [switch]
+    $nightlies
+  )
+  
+  [uri]$base_server_uri =  '<%= locals[:download_url] %>' # 'https://www.chef.io/chef/' 
+  
+  function New-Uri {
+    param ($baseuri, $newuri)
+    
+    new-object System.Uri $baseuri, $newuri
+  }
+  
+  function Get-WebContent {
+    param ($uri, $filepath)
+    $proxy = New-Object -TypeName System.Net.WebProxy
+    $wc = new-object System.Net.WebClient
+    $proxy.Address = $env:http_proxy
+    $wc.Proxy = $proxy
+    
+    try { 
+      if ([string]::IsNullOrEmpty($filepath)) {
+        $wc.downloadstring($uri)
+      }
+      else {
+        $wc.downloadfile($uri, $filepath)
+      }
+    }
+    catch {
+      $exception = $_.Exception
+      Write-Host "There was an error: "
+      do {
+        Write-Host "`t$($exception.message)"
+        $exception = $exception.innerexception
+      } while ($exception)
+      throw "Failed to download from $uri."
```

Comment:
Is it possible to pass just file_path and a blank uri to this?  It would be nice to allow that here because I can then port this directly into https://github.com/chef/knife-windows/blob/master/lib/chef/knife/bootstrap/windows-chef-client-msi.erb